app/models/detailed_guide.rb

```diff
@@ -62,6 +65,35 @@ def related_detailed_guide_ids=(detailed_guide_ids)
     self.related_documents = other_related_documents + detailed_guide_documents
   end
 
+  def need_ids
+    return @need_ids if @need_ids
+    return [] unless document # TODO: Remove
+
+    response = Whitehall.publishing_api_v2_client.get_links(
+      document.content_id
+    )
+
+    return unless response
+
+    @need_ids = response["links"]["meets_user_needs"]
+  end
+
+  def patch_meets_user_needs_links
+    return unless @need_ids
```

Comment:
Actually, now that the test suite has finished running, there are quite a few new errors somewhat related to this. If the `patch_meets_user_needs_links` method uses `need_ids` rather than accessing the value directly, if you just have an Edition, and you save it, `patch_meets_user_needs_links` might lookup the current needs by calling `need_ids` and then proceed to call out to the Publishing API.

However, If you access the value directly, you avoid the possibility that what you are considering saving has actually just been fetched from the Publishing API. I've changed this back for now, what do you think?